docs/projects/project_based/template/index.rst

```diff
@@ -0,0 +1,186 @@
+
+\**\* MENTION IF ANY MODES ARE AVAILABLE FOR TESTS \**\*
+
+The following test program files are available:
+
+=============== ==========================================
+Test program    Usage
+=============== ==========================================
+test_program_si Tests the parallel interface capabilities.
+test_program_pi Tests the serial interface capabilities.  
+=============== ==========================================
+
+Available configurations & tests combinations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+============= =============== ===================================
+Configuration Test            Build command 
+============= =============== ===================================
+cfg_si        test_program_si make CFG=cfg_si TST=test_program_si
+cfg_pi        test_program_pi make CFG=cfg_pi TST=test_program_pi 
+============= =============== ===================================
+
+.. warning::
+    Mixing a wrong pair of CFG and TST will result in a building errror. 
+    Please checkout the proposed combinations before running a custom test.
+    
+Clock scheme
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CPU/Memory interconnects addresses
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Interrupts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Building the test bench
+-------------------------------------------------------------------------------
+
+The design is built upon ADI's generic HDL reference design framework.
+ADI does not distribute the bit/elf files of these projects so they
+must be built from the sources available :git-hdl:`here </>`. To get
+the source you must
+`clone <https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository>`__
+the HDL repository, and then build the project as follows:.
+
+**Linux/Cygwin/WSL**
+
+*Example 1*
+
+Build all the possible combinations of tests and configurations, using only the
+command line.
+
+.. code-block::
+   :linenos:
+
+   user@analog:~$ cd testbenches/ad7616
+   user@analog:~/hdl/projects/ad7616$ make
+
+*Example 2*
+
+Build all the possible combinations of tests and configurations, using the
+Vivado GUI. This command will launch Vivado, will run the simulation and display
+the waveforms.
+
+.. code-block::
+   :linenos:
+
+   user@analog:~$ cd testbenches/ad7616
+   user@analog:~/hdl/projects/ad7616/$ make MODE=gui
+
+*Example 3*
+
+Build a particular combination of test and configuration, using the Vivado GUI.
+This command will launch Vivado, will run the simulation and display the
+waveforms.
+
+.. code-block::
+   :linenos:
+
+   user@analog:~$ cd testbenches/ad7616
+   user@analog:~/hdl/projects/ad7616$ make MODE=gui CFG=cfg_pi TST=test_program_pi
+
+The result of the build, if parameters were used, will be in the ``runs`` folder
+named by the configuration used:
+
+if the following command was run
```

Comment:
The built projects can be found in the ``runs`` folder, where each configuration specific build has it's own folder named after the configuration file's name. Example: if the following command was run for a single configuration in the clean folder (no ``runs`` folder available):